### DISCUSSION TITLE
Sorting Options for single Column

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#2673] [Ideas]
I don't think there is a way to currently do this but wanted to raise it as a possible addition. I have a resource which has "first_name" and "surname" fields. I also have an accessor on my model "name" which simply concatenates the two. It would be useful to have the option to be able to sort by "first_name" and also sort by "surname" but keep the accessor field "name" as a column without having to resort to using "first_name" and "surname" as separate columns. This mainly for readability.

**Comment by ManojKiranA** [comment]
Pls check out the [sorting](https://filamentphp.com/docs/2.x/tables/columns#sorting) docs

You can do something like this

`TextColumn::make('name')->sortable(['first_name', 'surname'])`

Or 

```
TextColumn::make('name')
    ->sortable(query: function (Builder $query, string $direction): Builder {
        return $query
            ->orderBy('first_name', $direction)
            ->orderBy('surname', $direction);
    })
```

**Reply by titantwentyone** [reply]
This sorts the column using two fields. It doesn't provide a choice to the user to sort by either "first_name" or "surname".

**Reply by danharrin** [reply]
I doubt this will ever be a feature, I can see the use case but I don't think it has enough other uses for the amount of work it would take to implement.

**Reply by titantwentyone** [reply]
No worries. Just thought I'd throw it out there.

**Reply by ManojKiranA** [reply]
can you tell me what do you have in mind @titantwentyone . But like @danharrin said this is a hell of a feature to implement.

**Reply by titantwentyone** [reply]
@ManojKiranAppathurai Just really what I outlined really. In terms of UI, I imagined something similar to how the filters are laid out and have custom sorts in there as a list.

**Reply by aksalsf** [reply]
How to implement this?

```
TextColumn::make('name')
    ->sortable(query: function (Builder $query, string $direction): Builder {
        return $query
            ->orderBy('first_name', $direction)
            ->orderBy('surname', $direction);
    })
```

The sortable looks only accept bool | array

**Reply by danharrin** [reply]
`->sortable(['first_name', 'surname'])`

**Reply by aksalsf** [reply]
Is it possible to create custom query for sort?

**Reply by danharrin** [reply]
Your code example looks correct

**Reply by mrgla55** [reply]
Had this come up as a requirement for me too and the replies confirmed what I thought.

Could it be done with a toggle on the table with something like "sortHiddenColumns", which would show the sort column (when using Split) regardless of hidden status? 

I did find a workaround using hiddenFrom('sm') breakpoint on a column within a Split. Still shows in the column in sort dropdown when hidden. But means the extra columns will show if they're on small mobile screens (which isnt a problem for me)
